awx/main/models/events.py

```diff
@@ -719,34 +640,13 @@ class Meta:
     def get_absolute_url(self, request=None):
         return reverse('api:ad_hoc_command_event_detail', kwargs={'pk': self.pk}, request=request)
 
-    def save(self, *args, **kwargs):
-        # If update_fields has been specified, add our field names to it,
-        # if it hasn't been specified, then we're just doing a normal save.
-        update_fields = kwargs.get('update_fields', [])
+    def _update_from_event_data(self):
```

Comment:
`bulk_create` doesn't actually *call* `.save()`, so I've changed this method to be something we can call manually to update the Django ORM objects *before* they're passed to `bulk_create`.

Reply:
`copy_ansible_runner_on_event_status_into_tower_runner_on_event_status` or some other, more descriptive function name.

Reply:
Organizationally, I see how it's a bit confusing, see discussion on `host_name` elsewhere here. That field was promoted from the "Ansible" event data to the "runner" event data in the callback receiver before the message was sent. It would be better to have all these promotion tasks put into the same place ("copy" in your wording). But that organizational issue won't be helped by improved naming. So if we're not refactoring, I would leave it as-is.

Also, the logging here is a really important feature, and I would certainly like to separate that from the other actions, for testing and other things.

Reply:
Agreed, all of these functions and inheritance in `models/events.py` are really starting to show how complex they've grown.  I wouldn't mind refactoring it to clean it up some, but I'm also not interested in doing it _right now_.